### PR TITLE
[SYCL] Add Property interface to 4 runtime classes

### PR DESCRIPTION
**Pull request by lbushi25**
This PR adds the property interface functions get_property and has_property to host_sampled_image_accessor, host_unsampled_image_accessor, sampled_image_accessor and unsampled_image_accessor runtime classes.

### CODE REVIEW

File: sycl/include/sycl/accessor.hpp [hunk 1]
```diff
@@ -3763,6 +3765,20 @@ class __SYCL_EBO unsampled_image_accessor :
   }
 
   /* -- property interface members -- */
+  template <typename Property> bool has_property() const noexcept {
```
Comment (lbushi25) [inline sycl/include/sycl/accessor.hpp:3768]: Property interface for unsampled_image_accessor

File: sycl/include/sycl/accessor.hpp [hunk 2]
```diff
@@ -3888,6 +3903,20 @@ class __SYCL_EBO host_unsampled_image_accessor
   }
 
   /* -- property interface members -- */
```
Comment (lbushi25) [inline sycl/include/sycl/accessor.hpp:3905]: Property interface for host_unsampled_image_accessor.

File: sycl/include/sycl/accessor.hpp [hunk 3]
```diff
@@ -4034,6 +4063,20 @@ class __SYCL_EBO sampled_image_accessor :
...
+  template <typename Property> bool has_property() const noexcept {
```
Comment (lbushi25) [inline sycl/include/sycl/accessor.hpp:4066]: Property interface for sampled_image_accessor

File: sycl/include/sycl/accessor.hpp [hunk 4]
```diff
@@ -4134,6 +4177,20 @@ class __SYCL_EBO host_sampled_image_accessor
   }
 
   /* -- property interface members -- */
```
Comment (lbushi25) [inline sycl/include/sycl/accessor.hpp:4179]: Property interface for host_sampled_image_accessor.

**Review by KornevNikita — COMMENTED** [review]
LGTM.

BTW clang-format only formats the modified code, is it intended to format the rest?
I'd rather not format to preserve the git-blame history, but that's at your discretion.

Fail on windows: https://github.com/intel/llvm/issues/12797

**Comment by lbushi25** [comment]
> LGTM.
> 
> BTW clang-format only formats the modified code, is it intended to format the rest? I'd rather not format to preserve the git-blame history, but that's at your discretion.
> 
> Fail on windows: #12797

I usually just run Ctrl + K afterwards on vim after modifying a file and sometimes it also changes other parts of code. I will quickly revert the formatting the changes as I agree about preserving the history. Thanks!

**Comment by lbushi25** [comment]
Windows Failures: #12797

**Comment by lbushi25** [comment]
@againull friendly ping

File: sycl/include/sycl/accessor.hpp [hunk 5]
```diff
@@ -3763,6 +3763,20 @@ class __SYCL_EBO unsampled_image_accessor :
...
+    return false;
+#endif
+  }
+  template <typename Property> Property get_property() const {
+#ifndef __SYCL_DEVICE_ONLY__
+    return getPropList().template get_property<Property>();
+#else
+    return Property();
+#endif
+  }
```
[the remark this answers is not in the document]
Comment (lbushi25) [reply sycl/include/sycl/accessor.hpp:3779]: Thanks for suggestion, I'm on it!

[the remark this answers is not in the document]
Comment (KornevNikita) [reply sycl/include/sycl/accessor.hpp:3779]: > Shouldn't this implementation be similar to what we have in the accessor? https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/accessor.hpp#L2423-L2463
> 
> I.e. different methods for runtime properties and compile-time properties.

It was made so according to https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_accessor_properties.asciidoc

As I understand this spec only affects base accessor, so no changes for another accessor-* classes are required, isn't it?

[the remark this answers is not in the document]
Comment (againull) [reply sycl/include/sycl/accessor.hpp:3779]: Unfortunately I don't know the answer about what this specification should affect. But let me rephrase my question: is proposed implementation going to work properly for compile-time properties?

[the remark this answers is not in the document]
Comment (KornevNikita) [reply sycl/include/sycl/accessor.hpp:3779]: As I understand, only base accessor accepts `accessor_property_list`, so only base accessor can accept compile-time properties

[the remark this answers is not in the document]
Comment (againull) [reply sycl/include/sycl/accessor.hpp:3779]: Ok, I see, thanks for clarifying.

**Review by againull — APPROVED** [review]

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers ping
